config/moved_config.go

```diff
@@ -147,9 +147,20 @@ func (cfg *Config) validateService() error {
 			}
 		}
 	}
+
 	return nil
 }
 
+// since mustValidateService will be used for next version, so use nolint to escape the `unused` lint error
+// nolint
+func (cfg *Config) mustValidateService() error {
+	if err := cfg.validateService(); err != nil {
+		return err
+	}
+
+	return cfg.Telemetry.Validate()
```

Comment:
Shouldn't this be part of the main Validate function? Why do you need this in a specific extra function (mustValidateService)?

Reply:
This also follows [example 2](https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#example-2---changing-the-return-values-of-a-function), since I introduced a new validating rule for config file, which may result that the collector with wrong config file crashes when upgraded to new version if the pr is merged. 
So I first mark old validation function as deprecated and use the new validation for next release.

Reply:
Oh, I see the confusion now. The example shows how one would proceed in changing the signature of a function to add a new return value (an error).

In your case, you should just add the new condition to the function. But instead of returning an error, you should just log a warning for a couple of releases and then return an error. In this case, given the bug is because the validation does not exist for this condition, adding the condition wouldn't be considered a breaking change (even though it would indeed break already faulty instances).

cc @open-telemetry/collector-contrib-approvers, for more opinions on this.

Reply:
> Oh, I see the confusion now. The example shows how one would proceed in changing the signature of a function to add a new return value (an error).
> 
> In your case, you should just add the new condition to the function. But instead of returning an error, you should just log a warning for a couple of releases and then return an error. In this case, given the bug is because the validation does not exist for this condition, adding the condition wouldn't be considered a breaking change (even though it would indeed break already faulty instances).
> 
> cc @open-telemetry/collector-contrib-approvers, for more opinions on this.

Ok, I got it. So I need to do following things
- remove `mustValidateService` function, just add new validation to original `validateService` function
- for next release(N+1), just log a warning without return error, so the instance with faulty config can still start.
- for N+2 release, return error for this validation. 

@jpkrohling Do I understand this correctly?

Reply:
Yes, sounds good to me.

Reply:
@jpkrohling  I already remove `mustValidateService` and move the validation of Telemetry config directly into `validateService ` function, and in validateService function I initialize a zap.logger to log for the validation fail. Please help review this, thank you =D